pkg/cpio/newc/newc.go

```diff
@@ -118,12 +119,21 @@ func (w *writer) WriteRecord(f cpio.Record) error {
 	buf := &bytes.Buffer{}
 	hdr := headerFromInfo(f.Info)
 	hdr.CRC = 0
+
+	// in newc, the FileSize is 0 for all but LNK and REG.
+	if f.Info.Mode&syscall.S_IFMT != syscall.S_IFREG &&
+		f.Info.Mode&syscall.S_IFMT != syscall.S_IFLNK {
+		hdr.FileSize = 0
```

Comment:
no. Because you may be reading an archive that incorrectly just passes on the non-zero size, even though only IFREG and IFLNK have size. This is a pretty cheap self-protection device.